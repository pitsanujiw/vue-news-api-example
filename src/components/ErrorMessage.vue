<template>
  <div
    class="d-flex align-center mx-auto justify-center"
    style="height: calc(100vh - 125px)"
  >
    <div>
      <div class="mr-10 ml-10 d-flex justify-center text-h7 text-center">
        Opps! The page may be broken, or the page may have been error.
      </div>
      <div class="mr-10 ml-10 d-flex justify-center text-h7 text-center">
        Check to see if the page you're trying to reload again.
      </div>
      <div class="d-flex mt-10 justify-center">
        <v-btn color="primary" @click="resetAndFetch">Reload</v-btn>
      </div>
    </div>
  </div>
</template>

<script>
import { mapMutations, mapActions } from 'vuex';

import { RESET_STATE } from '@/constants';

export default {
  name: 'ErrorMessage',
  methods: {
    ...mapMutations({
      resetState: RESET_STATE,
    }),
    ...mapActions(['newsTopHighlight']),
    async resetAndFetch() {
      this.resetState();
      await this.newsTopHighlight();
    },
  },
};
</script>

<style>
</style>
