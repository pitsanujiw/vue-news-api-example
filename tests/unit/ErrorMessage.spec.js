import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import ErrorMessage from '../../src/components/ErrorMessage.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('ErrorMessage', () => {
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Renders ErrorMessage Component', () => {
    const wrapper = shallowMount(ErrorMessage, { store, localVue, vuetify });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
