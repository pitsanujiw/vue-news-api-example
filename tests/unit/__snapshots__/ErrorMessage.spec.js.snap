// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Renders ErrorMessage Component 1`] = `
<div class="d-flex align-center mx-auto justify-center" style="height: calc(100vh - 125px);">
  <div>
    <div class="mr-10 ml-10 d-flex justify-center text-h7 text-center">
      Opps! The page may be broken, or the page may have been error.
    </div>
    <div class="mr-10 ml-10 d-flex justify-center text-h7 text-center">
      Check to see if the page you're trying to reload again.
    </div>
    <div class="d-flex mt-10 justify-center">
      <v-btn-stub dark="true" tag="button" activeclass="" type="button">Reload</v-btn-stub>
    </div>
  </div>
</div>
`;
